File: 프로그래머스/1/12928. 약수의 합/약수의 합.java
import java.util.*;

class Solution {
    public int solution(int n) {
        if(n <= 1) return n;
        int num = 1;
        int acc = 0;
        while(num < Math.sqrt(n)) {
            if(n % num == 0) {
                acc += num;
                acc += n/num;
            }
            num++;
        }
        if(num == Math.sqrt(n)) {
            acc += num;
        }

        return acc;
    }
}

File: 프로그래머스/1/12928. 약수의 합/약수의 합.kt
class Solution {
    fun solution(n: Int): Int {
        var answer = 0
        val s = Math.sqrt(n.toDouble()).toInt()
        for(i in 1..s) {
            if(n%i==0) {
                answer += i
                answer += n/i
            }
        }
        
        if(s*s == n) {
            answer-=s
        }
        return answer
    }
}